src/Statement/Grouper/FilteringStatementGrouper.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use InvalidArgumentException;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class FilteringStatementGrouper implements StatementGrouper {
+
+	/**
+	 * @var StatementList[]
+	 */
+	private $groups = array();
```

Comment:
What did Jonas do? I'm confused. The additional commit does have your avatar, not Jonas'.